src/main/webapp/resources/js/pages/launch/components/UploadReferenceFile.jsx

```diff
@@ -0,0 +1,40 @@
+import React from "react";
+import { message } from "antd";
+import { setBaseUrl } from "../../../utilities/url-utilities";
+import { useLaunchDispatch } from "../launch-context";
+import { DragUpload } from "../../../components/files/DragUpload.jsx";
+
+/**
+ * Component to upload a reference file to the launch pipeline page.
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export function UploadReferenceFile() {
+  const { dispatchReferenceFileUploaded } = useLaunchDispatch();
+
+  const options = {
+    multiple: true,
+    showUploadList: false,
+    action: setBaseUrl(`/ajax/reference-files`),
+    onChange(info) {
+      const { status } = info.file;
+      if (status === "done") {
+        message.success(i18n("UploadReferenceFile.success", info.file.name));
+        dispatchReferenceFileUploaded({
+          name: info.file.name,
+          id: info.file.response.id,
+        });
+      } else if (status === "error") {
+        message.error(`${info.file.name} file upload failed.`);
```

Comment:
Internationalize

Reply:
Updated in a10eeec132bd8397805a30a186862baf0ab6a576